File: android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/components/styles/sources/AbstractSourceConsumer.java
package com.mapbox.rctmgl.components.styles.sources;

import android.content.Context;

import com.mapbox.rctmgl.components.AbstractMapFeature;

public abstract class AbstractSourceConsumer extends AbstractMapFeature {
    public AbstractSourceConsumer(Context context) {
        super(context);
    }

    abstract public String getID();
}

File: android/src/main/java/com/rnmapbox/rnmbx/components/styles/sources/AbstractSourceConsumer.kt
package com.rnmapbox.rnmbx.components.styles.sources

import android.content.Context
import com.rnmapbox.rnmbx.components.AbstractMapFeature

abstract class AbstractSourceConsumer(context: Context?) : AbstractMapFeature(context) {
    abstract val iD: String?
}